src/lib.rs

```diff
@@ -4,11 +4,155 @@
 #![feature(thread_local)]
 #![allow(unused_imports)]
 #![feature(hash_extract_if)]
+#![allow(dead_code)]
 
 // interface and safeposix are public because otherwise there isn't a great
 // way to 'use' them for benchmarking.
 pub mod interface;
-// mod lib_fs_utils;
 pub mod safeposix;
 pub mod tests;
 pub mod fdtables;
+
+use std::sync::{Condvar, Mutex};
+
+use crate::safeposix::dispatcher::*;
+
+#[macro_export]
+macro_rules! dispatch {
+    ($cageid:expr, $callnum:expr) => {
+        dispatcher(
+            $cageid, $callnum, BLANKARG, BLANKARG, BLANKARG, BLANKARG, BLANKARG, BLANKARG,
+        )
+    };
+    ($cageid:expr, $callnum:expr, $arg1:expr) => {
+        dispatcher(
+            $cageid, $callnum, $arg1, BLANKARG, BLANKARG, BLANKARG, BLANKARG, BLANKARG,
+        )
+    };
+    ($cageid:expr, $callnum:expr, $arg1:expr, $arg2:expr) => {
+        dispatcher(
+            $cageid, $callnum, $arg1, $arg2, BLANKARG, BLANKARG, BLANKARG, BLANKARG,
+        )
+    };
+    ($cageid:expr, $callnum:expr, $arg1:expr, $arg2:expr, $arg3:expr) => {
+        dispatcher(
+            $cageid, $callnum, $arg1, $arg2, $arg3, BLANKARG, BLANKARG, BLANKARG,
+        )
+    };
+    ($cageid:expr, $callnum:expr, $arg1:expr, $arg2:expr, $arg3:expr, $arg4:expr) => {
+        dispatcher(
+            $cageid, $callnum, $arg1, $arg2, $arg3, $arg4, BLANKARG, BLANKARG,
+        )
+    };
+    ($cageid:expr, $callnum:expr, $arg1:expr, $arg2:expr, $arg3:expr, $arg4:expr, $arg5:expr) => {
+        dispatcher(
+            $cageid, $callnum, $arg1, $arg2, $arg3, $arg4, $arg5, BLANKARG,
+        )
+    };
+    ($cageid:expr, $callnum:expr, $arg1:expr, $arg2:expr, $arg3:expr, $arg4:expr, $arg5:expr, $arg6:expr) => {
+        dispatcher($cageid, $callnum, $arg1, $arg2, $arg3, $arg4, $arg5, $arg6)
+    };
+}
+
+pub fn lind_lindrustinit(verbosity: i32) {
+    unsafe {
+        lindrustinit(verbosity as isize);
+    }
+}
+
+pub fn lind_lindrustfinalize() {
+    unsafe {
+        lindrustfinalize();
+    }
+}
+
+pub fn lind_rustposix_thread_init(cageid: u64, signalflag: u64) {
+    unsafe {
+        rustposix_thread_init(cageid, signalflag);
+    }
+}
+
+pub fn lind_write_inner(fd: i32, buf: *const u8, count: usize, cageid: u64) {
```

Comment:
I believe this function can be removed

Reply:
Most of the stuff added in src/lib.rs is something from original rustposix lib.rs under wasmtime. It is something already there before I joined the wasm team, so I am not very sure what's their original purpose. But looks like they are no longer useful so I'll clean these up later

Reply:
OK yeah this still needs to be done. Not sure when these were added but this should all be handled via the dispatcher i believe.